### PR TITLE
<Input /> - Use compound components for prefix/suffix rendering

### PR DESCRIPTION
**Pull request by NitayRabi**
### 🔦 Summary
1. Add `CustomAffix` component to replace rendering of `unit` and support text rendering. (used when suffix/prefix is string only)
2. Add `IconAffix` for correct rendering of icons (replacing magnifyingGlass prop)
3. Match design with definition in specs (Remove separator, change paddings to fit constraints)
4. Deprecate (log only) magnifyingGlass, unit prop and Input.Unit component.
5. Remove redundant stories, create Test page and e2e tests for new Input Applitool baseline.

### ✅ Checklist
- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [x] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [x] Story
  - [x] API description
  - [ ] Other (explain)
- 🔬 Change is tested
  - [x] Component
  - [x] Visual test

### CODE REVIEW

File: src/Input/CustomAffix/CustomAffix.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import styles from './CustomAffix.scss';
+import InputConsumer from '../InputConsumer';
+
+const CustomAffix = ({ children, value }) => (
```
Comment (argshook) [inline src/Input/CustomAffix/CustomAffix.js]: @NitayRabi could it be named just `affix`?

Comment (argshook) [reply src/Input/CustomAffix/CustomAffix.js]: i think yes, `custom` doesn't really explain anything either imo

Comment (NitayRabi) [reply src/Input/CustomAffix/CustomAffix.js]: NP

File: src/Input/InputSuffix.js [hunk 2]
```diff
@@ -94,6 +98,7 @@ const InputSuffix = ({
     {
       component: () => (
         <div
+          size={size}
```
Comment (shlomitc) [inline src/Input/InputSuffix.js]: looks redudnant

File: src/Input/InputSuffix.js [hunk 3]
```diff
@@ -83,6 +86,7 @@ const InputSuffix = ({
       component: () => (
         <ThemedInputHelpSuffix
           theme={theme}
+          size={size}
```
Comment (shlomitc) [inline src/Input/InputSuffix.js]: are you sure the component is using it?

File: src/Input/Ticker/Ticker.js [hunk 4]
```diff
@@ -4,22 +4,29 @@ import classnames from 'classnames';
...
-    <div
-      className={classnames(styles.down, { [styles.disabled]: downDisabled })}
-      onClick={downDisabled ? null : onDown}
-    >
-      <FormFieldSpinnerDown />
-    </div>
-  </div>
+  <InputConsumer consumerCompName={Ticker.displayName}>
+    {props => (
+      <div className={styles.root} data-hook="ticker">
```
Comment (shlomitc) [inline src/Input/Ticker/Ticker.js]: data-hook should come from the parent via `dataHook`

File: src/Input/Unit/Unit.js [hunk 5]
```diff
@@ -1,13 +1,25 @@
...
+  constructor(props) {
+    super(props);
+    deprecationLog(
+      'Input.Unit component is deprecated and will be removed in the next major release, please use Input.Affix instead',
+    );
+  }
+  render() {
+    const { children, value } = this.props;
+    return (
+      <div className={styles.unit} data-hook="unit">
```
Comment (shlomitc) [inline src/Input/Unit/Unit.js]: same `data-hook` issue

File: stories/TextField/ExampleAdornments.js [hunk 6]
```diff
@@ -4,6 +4,8 @@ import PropTypes from 'prop-types';
 import { Layout, Cell } from 'wix-style-react/Layout';
 import Input from 'wix-style-react/Input';
 import FormField from 'wix-style-react/FormField';
+import Date from '../../src/new-icons/Date';
```
Comment (shlomitc) [inline stories/TextField/ExampleAdornments.js]: `wix-style-react/new-icons/Date`

File: stories/TextField/ExampleNumberInput.js [hunk 7]
```diff
@@ -65,6 +65,9 @@ export default class ExampleNumberInput extends React.Component {
           max={max}
           {...errorProps}
           suffix={
+            // <Input.Group>
```
Comment (shlomitc) [inline stories/TextField/ExampleNumberInput.js]: remove

File: src/Input/Input.js [hunk 8]
```diff
@@ -194,11 +225,12 @@ class Input extends Component {
...
               tooltipPlacement={tooltipPlacement}
               onTooltipShow={onTooltipShow}
             />
-          </div>
-        )}
+          )}
+        </InputContext.Provider>
       </div>
     );
   }
```
Comment (shlomitc) [inline src/Input/Input.js]: remove `magnifyingGlass` and `unit` from proptypes

**Comment by NitayRabi** [comment]
> @NitayRabi
> 
> * [ ]  `Size: Large` text size doesn't match the guidelines
> * [ ]  When `Size: Large` and `RoundInput` props are enabled, the corners aren't fully round.
> * [ ]  `prefix` and `unit` props text weight doesn't match the guidelines
> * [ ]  When `disabled` the component the hover effect should be disabled
> * [ ]  Where the `help` prop comes from? It's not in the specs
> * [ ]  `TooltipPlacement` should support `<popover/>` component's auto positioning
> * [ ]  if `magifyingGlass` prop is for `<Search/>` component, then the icons position is not right. [Searches specs](https://zpl.io/2pwonyb)
> * [ ]  same applies to `clearButton` as for magnifyingGlass

Regarding the disabled -> hover relationship - #2893 will fix it.
Regarding `<Search />` + Clear button component - could you elaborate what conflicts the spec? (Under Search component story the icons seems aligned, search icon can be used as an IconAffix, but it's not the search component)
Regarding `help` property - @shlomitc should it be deprecated? used in composition with FormField?

**Comment by NitayRabi** [comment]
> @NitayRabi requests look fixed.
> Does the `magifyingGlass` prop was also deprecated? (If yes, it's good)

Yes! you now use compound components to achieve this behaviour

**Comment by mykas** [comment]
Hey @NitayRabi , whats your rough estimation for this PR?

**Pull request merged**